x-pack/plugins/secops/public/components/page/flyout.tsx

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { EuiFlyout, EuiFlyoutBody, EuiFlyoutHeader, EuiText, EuiTitle } from '@elastic/eui';
+import * as React from 'react';
+import styled from 'styled-components';
+
+export const Overlay = styled.div`
+  position: absolute;
+  top: 15%;
+  right: 0%;
+  width: 30px;
+  z-index: 1;
+  height: 60%;
+`;
+
+export const Button = styled.div`
+  padding: 10px 0 10px 0;
+  display: flex;
+  position: absolute;
+  top: 30%;
+  right: 5%;
+  width: 100%;
+  z-index: 2;
+  justify-content: center;
+  text-align: center;
+  border-top: 1px solid #c5c5c5;
+  border-bottom: 1px solid #c5c5c5;
+  border-left: 1px solid #c5c5c5;
+  border-radius: 6px 0 0 6px;
+  box-shadow: 0 3px 3px -1px rgba(173, 173, 173, 0.5), 0 5px 7px -2px rgba(173, 173, 173, 0.5);
+  background-color: #fff;
+`;
+
+export const Text = styled(EuiText)`
+  width: 12px;
+  z-index: 3;
+`;
+
+interface Props {
+  children?: React.ReactNode;
+}
+
+interface State {
+  isFlyoutVisible: boolean;
+}
+
+type SetState = (opts: State) => void;
+
+export const showFlyout = (isFlyoutVisible: boolean, setState: SetState) =>
+  setState({ isFlyoutVisible });
+
+export const closeFlyout = (setState: SetState) => showFlyout(false, setState);
+
+export const openFlyout = (setState: SetState) => showFlyout(true, setState);
+
+export const showFlyoutElement = (
```

Comment:
I'm curious: why did you structure this code as "more of a function" than a typical `SFC`, using `pure`, etc?

If I understand the usage correctly, the former is not required, but perhaps I'm missing something?